applications/asset_tracker_v2/tests/cloud_codec_nrf/src/nrf_cloud_codec_test.c

```diff
@@ -0,0 +1,686 @@
+				&ui_buf,
+				&impact_buf,
+				&bat_buf,
+				1, 1, 1, 1, 1, 1, 1);
+	TEST_ASSERT_EQUAL(EXIT_SUCCESS, ret);
+	TEST_ASSERT_EQUAL(0, strncmp(MODEM_STATIC_BATCH_EXAMPLE,
+				     codec.buf,
+				     strlen(MODEM_STATIC_BATCH_EXAMPLE)));
+	TEST_ASSERT_FALSE(modem_stat_buf.queued);
+}
+
+void test_enc_batch_data_modem_dynamic(void)
+{
+	struct cloud_data_gnss gnss_buf = {0};
+	struct cloud_data_sensors sensor_buf = {0};
+	struct cloud_data_modem_static modem_stat_buf = {0};
+	struct cloud_data_modem_dynamic modem_dyn_buf = {
+		.band = 3,
+		.nw_mode = LTE_LC_LTE_MODE_NBIOT,
+		.rsrp = -8,
+		.area = 12,
+		.mccmnc = "24202",
+		.cell = 33703719,
+		.ip = "10.81.183.99",
+		.ts = 1000,
+		.queued = true,
+		.band_fresh = true,
+		.nw_mode_fresh = true,
+		.area_code_fresh = true,
+		.cell_id_fresh = true,
+		.rsrp_fresh = true,
+		.ip_address_fresh = true,
+		.mccmnc_fresh = true,
+	};
+	struct cloud_data_ui ui_buf = {0};
+	struct cloud_data_impact impact_buf = {0};
+	struct cloud_data_battery bat_buf = {0};
+
+	ret = cloud_codec_encode_batch_data(&codec,
+				&gnss_buf,
+				&sensor_buf,
+				&modem_stat_buf,
+				&modem_dyn_buf,
+				&ui_buf,
+				&impact_buf,
+				&bat_buf,
+				1, 1, 1, 1, 1, 1, 1);
+	TEST_ASSERT_EQUAL(EXIT_SUCCESS, ret);
+	TEST_ASSERT_EQUAL(0, strncmp(MODEM_DYNAMIC_BATCH_EXAMPLE,
+				     codec.buf,
+				     strlen(MODEM_DYNAMIC_BATCH_EXAMPLE)));
+	TEST_ASSERT_FALSE(modem_dyn_buf.queued);
+}
+
+void test_enc_batch_data_modem_dynamic_rsrp_too_small(void)
+{
+	struct cloud_data_gnss gnss_buf = {0};
+	struct cloud_data_sensors sensor_buf = {0};
+	struct cloud_data_modem_static modem_stat_buf = {0};
+	struct cloud_data_modem_dynamic modem_dyn_buf = {
+		.band = 3,
+		.nw_mode = LTE_LC_LTE_MODE_NBIOT,
+		.rsrp = INT16_MIN,
+		.area = 12,
+		.mccmnc = "24202",
+		.cell = 33703719,
+		.ip = "10.81.183.99",
+		.ts = 1000,
+		.queued = true,
+		.band_fresh = true,
+		.nw_mode_fresh = true,
+		.area_code_fresh = true,
+		.cell_id_fresh = true,
+		.rsrp_fresh = true,
+		.ip_address_fresh = true,
+		.mccmnc_fresh = true,
+	};
+	struct cloud_data_ui ui_buf = {0};
+	struct cloud_data_impact impact_buf = {0};
+	struct cloud_data_battery bat_buf = {0};
+
+	ret = cloud_codec_encode_batch_data(&codec,
+				&gnss_buf,
+				&sensor_buf,
+				&modem_stat_buf,
+				&modem_dyn_buf,
+				&ui_buf,
+				&impact_buf,
+				&bat_buf,
+				1, 1, 1, 1, 1, 1, 1);
+	TEST_ASSERT_EQUAL(-ENOMEM, ret);
+	TEST_ASSERT_FALSE(modem_dyn_buf.queued);
+}
+
+void test_enc_batch_data_sensor(void)
+{
+	struct cloud_data_gnss gnss_buf = {0};
+	struct cloud_data_sensors sensor_buf = {
+		.temperature = 23,
+		.humidity = 50,
+		.pressure = 101,
+		.bsec_air_quality = 50,
+		.env_ts = 1563968747123,
+		.queued = true,
+	};
+	struct cloud_data_modem_static modem_stat_buf = {0};
+	struct cloud_data_modem_dynamic modem_dyn_buf = {0};
+	struct cloud_data_ui ui_buf = {0};
+	struct cloud_data_impact impact_buf = {0};
+	struct cloud_data_battery bat_buf = {0};
```

Comment:
May be create global structure for these too so that you dont need to redeclare them and initialize them to {0} for every test case?  These declarations make the test look crowded. So its better to move out common parts elsewhere as much as possible.

Reply:
If C supported reinitalization of structs, I would do that. But without that, the result would look much worse.